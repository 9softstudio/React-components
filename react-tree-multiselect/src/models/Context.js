import React from 'react';

export default React.createContext({
    rootNode: {},
    nodes: [],
    onToggle: () => { },
    onCheck: () => { }
});