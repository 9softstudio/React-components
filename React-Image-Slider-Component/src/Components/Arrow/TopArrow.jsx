import React, { Component } from 'react';
import Context from '../../slider-context';

export default class TopArrow extends Component {
    render() {
        return (
            <Context.Consumer>{
                ({ onTopArrowClick }) =>
                    <img src="img/slider-top-arrow.png" className="slider-arrow-position top-arrow" onClick={onTopArrowClick} />
            }</Context.Consumer>);
    }
};