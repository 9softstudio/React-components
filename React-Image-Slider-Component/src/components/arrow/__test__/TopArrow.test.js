import React from 'react';
import TopArrow from '../TopArrow';
import Context from '../../../slider-context';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const component = renderer
        .create(
            <Context.Provider value={({ onTopArrowClick: jest.fn() })}>
                <TopArrow />
            </Context.Provider>
        );
    expect(component.toJSON()).toMatchSnapshot();
});

test('on click: onTopArrowClick()', () => {
    const onTopArrowClick = jest.fn();
    const component = renderer
        .create(
            <Context.Provider value={({ onTopArrowClick: onTopArrowClick })}>
                <TopArrow />
            </Context.Provider>
        );
    expect(component.toJSON().props.onClick).toEqual(onTopArrowClick);
});