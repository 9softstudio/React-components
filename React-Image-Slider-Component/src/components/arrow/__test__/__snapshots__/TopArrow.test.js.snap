// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  className="slider-arrow-position top-arrow"
  onClick={[MockFunction]}
  src="img/slider-top-arrow.png"
/>
`;
